lib/smack/Naming.cpp

```diff
@@ -178,16 +178,36 @@ bool Naming::isSmackGeneratedName(std::string n) {
 }
 
 std::string Naming::escape(std::string s) {
-  std::string Str(llvm::DOT::EscapeString(s));
+  std::string Str(s);
   for (unsigned i = 0; i != Str.length(); ++i)
     switch (Str[i]) {
-    case '\01':
-      Str[i] = '_';
-      break;
+    // Special boogie characters
     case '@':
       Str[i] = '.';
       break;
+    case '\01': case '\\':
     case ':':
+    case '(': case ')':
+    case '[': case ']':
+      Str[i] = '_';
+      break;
+    // Another character to escape would be '$', but SMACK internally
+    // generates LLVM IR that uses this character.
+    
+    // Reproduce behavior of llvm::DOT::EscapeString without backslashes
+    case '\n':
```

Comment:
Are we ever going to see `\n` and `\t` in names?
I really don't think so. So I suggest we remove that part. We do not have to faithfully model LLVM's EscapeString.